### PR TITLE
Add doctrine-project.json to root of the project.

### PR DESCRIPTION
**Pull request by jwage**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

#### Summary

Add doctrine-project.json to the root of the project for the website build process to read.

Related to https://github.com/doctrine/doctrine-website/pull/218

### CODE REVIEW

File: doctrine-project.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+    "active": true,
+    "name": "Migrations",
+    "shortName": "Migrations",
+    "slug": "migrations",
+    "docsSlug": "doctrine-migrations",
+    "composerPackageName": "doctrine/migrations",
```
Comment (Majkl578) [inline doctrine-project.json]: Can't this be inferred from `composer.json`?

File: doctrine-project.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "docsRepositoryName": "migrations",
+    "docsPath": "/docs",
+    "description": "PHP Doctrine Migrations project offer additional functionality on top of the database abstraction layer (DBAL) for versioning your database schema and easily deploying changes to it. It is a very easy to use and a powerful tool.\n",
```
Comment (Majkl578) [inline doctrine-project.json]: Can't this be inferred from `composer.json`?

File: doctrine-project.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "keywords": [
```
Comment (Majkl578) [inline doctrine-project.json]: Can't this be inferred from `composer.json`?

**Comment by Majkl578** [comment]
Does this mean `master` always has to hold authoritative information about all versions?

**Comment by Majkl578** [comment]
I'd prefer automating both inferring data from `composer.json` and making autodiscovery of the versions (from existing branches, from `branch-alias`). This adds one more burden for new releases. :/

**Comment by jwage** [comment]
We can definitely work towards that as a goal. Should be a relatively easy enhancement.

**Comment by stof** [comment]
the drawback of automatic discovery of branches is that the doc website cares about *maintained* branches AFAIK, and github does not have a way to mark branches as unmaintained.

**Comment by stof** [comment]
I'm clearing the milestone, as this is not really tied to any release of the package

**Comment by Majkl578** [comment]
> and github does not have a way to mark branches as unmaintained.

That is true. We could solve this for example by `.not-maintained` placeholder file.
With the approach here, we would end up with different json in each branch and also additional thing to bump every minor/major release in every project.

**Comment by stof** [comment]
@Majkl578 the project file does not only list the branches. It also defines a stable version, and marks some versions as upcoming. All these might be hard to read from composer.json (and even harder when you consider it comes from *multiple* composer.json, which then might disagree).
Considering that the file in the default branch of the repo is the one taken into account by the website makes sense to me, as the website is working with the repo, not with a given revision of the repo).
The placeholder file might not be good for performance of the builder though, as you would have to check all git branches for this file when building it, instead of knowing upfront which branch you care about.

**Comment by jwage** [comment]
@stof @Majkl578 It is definitely a challenge to have a different doctrine-project.json for each version. We already have to update the doctrine-website repo when we release new things, so this is at least a step forward to where each project can make the change in their own repo instead of having to make the change in the website code.

**Pull request merged**